### PR TITLE
Add EdgeX reST documentation

### PR DESCRIPTION
**Pull request by andyf1967**
Signed-off-by: steve osselton <steve@iotechsys.com>

### CODE REVIEW

File: docs/build.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+cp ../core/data/raml/*.raml ./core/data/
+cp ../core/metadata/raml/*.raml ./core/metadata/
+cp ../core/command/raml/*.raml ./core/command/
+cp ../support/logging/raml/*.raml ./support/logging/
+cp ../export/client/raml/*.raml ./export/client/
+
+# Build image (copying in documentation sources)
+
+docker build -t doc-builder:latest -f Dockerfile.build .
+if [ -d _build ]
```
Comment (feclare) [inline docs/build.sh]: No need to test if _build exists, just run the `rm -rf`

File: docs/build.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
```
Comment (feclare) [inline docs/build.sh]: Add `set -e`, if some command fails the script should stop and report error

File: docs/build.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# Clean up
+
+rm ./core/data/*.raml
+rm ./core/metadata/*.raml
+rm ./core/command/*.raml
+rm ./support/logging/*.raml
+rm ./export/client/*.raml
+docker rm builder
+docker image prune -f
```
Comment (feclare) [inline docs/build.sh]: Not sure if we should prune docker images, may be some user does not want it

File: docs/build.sh (hunk 1 above)
Comment (SteveOss) [reply docs/build.sh]: OK, Will do.

File: docs/build.sh (hunk 2 above)
Comment (SteveOss) [reply docs/build.sh]: OK

File: docs/build.sh (hunk 3 above)
Comment (SteveOss) [reply docs/build.sh]: Have updated logic so that run container with --rm and use a bind mount to access built documentation, so this no longer needed.

**Pull request merged**